src/Common/Thermodynamics/states.jl

```diff
@@ -450,3 +508,24 @@ function PhaseNonEquil_pθq(
     e_int = internal_energy(param_set, T, q_pt)
     return PhaseNonEquil{FT, typeof(param_set)}(param_set, e_int, ρ, q_pt)
 end
+
+"""
+    PhaseNonEquil_ρpq(param_set, ρ, p, q_pt)
+
+Constructs a [`PhaseNonEquil`](@ref) thermodynamic state from:
+
+ - `param_set` an `AbstractParameterSet`, see the [`Thermodynamics`](@ref) for more details
+ - `ρ` density
+ - `p` pressure
+ - `q_pt` phase partition
+"""
+function PhaseNonEquil_ρpq(
+    param_set::APS,
+    ρ::FT,
+    p::FT,
+    q_pt::PhasePartition{FT},
+) where {FT <: Real}
+    T = air_temperature_from_ideal_gas_law(param_set, p, ρ, q_pt)
+    e_int = internal_energy(param_set, T, q_pt)
+    return PhaseNonEquil{FT, typeof(param_set)}(param_set, e_int, ρ, q_pt)
+end
```

Comment:
so you don't need saturation adjustment to get T for NonEquil ?

Reply:
Nope, no saturation adjustment required for `PhaseNonEquil`.